kroxylicious-filters/kroxylicious-oauthbearer-validation/src/main/java/io/kroxylicious/proxy/filter/oauthbearer/OauthBearerValidation.java

```diff
@@ -0,0 +1,119 @@
+import io.kroxylicious.proxy.plugin.PluginConfigurationException;
+import io.kroxylicious.proxy.plugin.Plugins;
+import io.kroxylicious.proxy.tag.VisibleForTesting;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+import static org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule.OAUTHBEARER_MECHANISM;
+
+@Plugin(configType = OauthBearerValidation.Config.class)
+public class OauthBearerValidation implements FilterFactory<OauthBearerValidation.Config, SharedOauthBearerValidationContext> {
+
+    @SuppressWarnings("unused")
+    @VisibleForTesting
+    public OauthBearerValidation() {
+        this.oauthHandler = new OAuthBearerValidatorCallbackHandler();
+    }
+
+    public OauthBearerValidation(OAuthBearerValidatorCallbackHandler oauthHandler) {
+        this.oauthHandler = oauthHandler;
+    }
+
+    private final OAuthBearerValidatorCallbackHandler oauthHandler;
+
+    static {
+        OAuthBearerSaslServerProvider.initialize();
+    }
+
+    @Override
+    @SuppressWarnings("java:S2245") // secure randomization not needed for exponential backoff
+    public SharedOauthBearerValidationContext initialize(FilterFactoryContext context, Config config) throws PluginConfigurationException {
+        Plugins.requireConfig(this, config);
+        Config configWithDefaults = new Config(
+                config.jwksEndpointUrl,
+                config.jwksEndpointRefreshMs() != null && config.jwksEndpointRefreshMs() >= 0L ? config.jwksEndpointRefreshMs()
+                        : SaslConfigs.DEFAULT_SASL_OAUTHBEARER_JWKS_ENDPOINT_REFRESH_MS,
+                config.jwksEndpointRetryBackoffMs() != null && config.jwksEndpointRetryBackoffMs() >= 0L ? config.jwksEndpointRetryBackoffMs()
+                        : SaslConfigs.DEFAULT_SASL_OAUTHBEARER_JWKS_ENDPOINT_RETRY_BACKOFF_MS,
+                config.jwksEndpointRetryBackoffMaxMs() != null && config.jwksEndpointRetryBackoffMaxMs() > 0L ? config.jwksEndpointRetryBackoffMaxMs()
+                        : SaslConfigs.DEFAULT_SASL_OAUTHBEARER_JWKS_ENDPOINT_RETRY_BACKOFF_MAX_MS,
+                config.scopeClaimName() != null && !config.scopeClaimName().trim().isEmpty() ? config.scopeClaimName()
+                        : SaslConfigs.DEFAULT_SASL_OAUTHBEARER_SCOPE_CLAIM_NAME,
+                config.subClaimName() != null && !config.subClaimName().trim().isEmpty() ? config.subClaimName() : SaslConfigs.DEFAULT_SASL_OAUTHBEARER_SUB_CLAIM_NAME,
+                config.authenticateBackOffMaxMs() != null && config.authenticateBackOffMaxMs() >= 0L ? config.authenticateBackOffMaxMs() : 60000L);
+        oauthHandler.configure(
+                createSaslConfigMap(configWithDefaults),
+                OAUTHBEARER_MECHANISM,
+                createDefaultJaasConfig());
+        LoadingCache<String, AtomicInteger> rateLimiter = Caffeine.newBuilder()
+                .expireAfterWrite(configWithDefaults.authenticateBackOffMaxMs(), TimeUnit.MILLISECONDS)
```

Comment:
I think we should place an upper bound on the size of the cache to limit the possible of a DoS against the proxy. Something like:

```suggestion
                .expireAfterWrite(configWithDefaults.authenticateBackOffMaxMs(), TimeUnit.MILLISECONDS)
                .maximumSize(1000)
```